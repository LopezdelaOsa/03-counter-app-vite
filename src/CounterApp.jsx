
import {useState} from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ( {value} ) => {
    const [counter, setCounter] = useState (value)

    /*function handleAdd (event) {*/
    const handleAdd = (event) => {
        /*console.log(event)
        value += 1
        console.log (value)*/

        //setCounter(counter + 1)
        setCounter ( (c) => c + 1 )
    }

    const handleSubtract = (event) => {
        setCounter (counter - 1)
    }

    const handleReset = (event) => setCounter(value)

    return (
        <>
            <h1>Counter - App</h1>
            <h2>{ counter }</h2>
            {/*<button onClick={ (event) => handleAdd(event) }>*/}
            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handleSubtract }> -1 </button>
            <button aria-label='boton-reset' onClick={ handleReset }> Inicio </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}