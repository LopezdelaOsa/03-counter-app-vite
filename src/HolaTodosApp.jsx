/*export function App() {
    //document.createElement....
    return (<h1>Hola a todos</h1>)
}
*/
//export default App (sin poner {} en el import)

export const HolaTodosApp = () => {
    return (
        <h1>Hola a todos !!</h1>
    )
}

// export const HolaTodosApp = () => <h1>En forma de función de flecha</h1>