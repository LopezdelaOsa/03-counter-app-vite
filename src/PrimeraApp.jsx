
//rafc

/*
import {Fragment} from 'react'

export const PrimeraApp = () => {
    return ( <Fragment>
        <h2>Primera APP de Raúl</h2>
        <p>Esto es el subtítulo</p>
        </Fragment> )
}*/

//const nuevoMensaje = [1,2,3,4]
const nuevoMensaje = {
    mensaje: 'Hola a todos',
    titulo: 'Raul'
}

const resultado = (a, b) => {
    return ('Esto lo que debe salir' + a + b)
}

export const PrimeraApp = () => {

    return ( <>
    {/*        <code> { JSON.stringify(nuevoMensaje) }</code>    */}
        <h1>{resultado(7,3)}</h1>
        <p>Esto es el subtitulo, sin fragmentos</p>
        </> )
}