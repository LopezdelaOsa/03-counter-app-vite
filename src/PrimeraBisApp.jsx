//rafc

/*
export const PrimeraBisApp = (props) => {

    console.log(props)

  return ( <>
    <h1>{props.titulo}</h1>
    <p>Aqui el subtitulo</p>
  </> )
}*/
import PropTypes from 'prop-types'
// Desestructurado:
export const PrimeraBisApp = ({titulo, subTitulo, nombre}) => {
/*
    if (!titulo) {
        throw new Error('El titulo es obligatorio')
    }
*/
    //console.log(props)

  return ( <>
    <h1 data-testid="test-titulo">{titulo}</h1>
    <p>{subTitulo}</p>
    <p>{subTitulo}</p>
    <p>{nombre}</p>
  </> )
}


PrimeraBisApp.propTypes = {
    titulo: PropTypes.string.isRequired,
    subTitulo: PropTypes.string
}

PrimeraBisApp.defaultProps = {
  nombre: 'Raul Lopezosa',
  //titulo: 'No hay titulo',
  subTitulo: 'No aparece el subtitulo'
}