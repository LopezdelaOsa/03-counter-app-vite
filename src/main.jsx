import React from 'react'
import ReactDOM from 'react-dom/client' 
//import {App} from './HolaTodosApp'
//import App from './HolaTodosApp' (sin llaves si el export es 'default' a todo el fichero)
//import { HolaTodosApp } from './HolaTodosApp'
//import { PrimeraApp } from './PrimeraApp'
import { PrimeraBisApp } from './PrimeraBisApp'
import './estilos.css'
import { CounterApp } from './CounterApp'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/*<PrimeraApp />*/}
        {/*<PrimeraBisApp titulo="La senda del tiempo" />*/}
        <CounterApp value={100} />
    </React.StrictMode>
)