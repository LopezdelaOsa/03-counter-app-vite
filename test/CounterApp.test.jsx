import { fireEvent, render, screen } from "@testing-library/react"
import {CounterApp} from  "../src/CounterApp"

describe('Pruebas en CounterApp', () => { 

    const valorInicial = 100

    test('Coincidencia con snapshot', () => { 
        const {container} = render(<CounterApp value={valorInicial}/>)
        expect(container).toMatchSnapshot()
    })

    test('Mostrar valor inicial: 100', () => { 
        render (<CounterApp value={valorInicial} />)
        expect(screen.getByText(100)).toBeTruthy()
        //expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain('100')
    })

    test('Aumento con boton +1', () => { 
        render (<CounterApp value={valorInicial} />)
        fireEvent.click(screen.getByText('+1'))
        expect(screen.getByText('101')).toBeTruthy()
    })

    test('Disminuir con boton +11', () => { 
        render (<CounterApp value={valorInicial} />)
        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByText('99')).toBeTruthy()
    })

    test('Iniciar el valor original', () => { 
        render (<CounterApp value={valorInicial} />)
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))

        // fireEvent.click(screen.getByText('Inicio'))
        // expect(screen.getByText(valorInicial)).toBeTruthy()
        fireEvent.click( screen.getByRole('button', {name: 'boton-reset'}) )
        expect(screen.getByText(valorInicial)).toBeTruthy()
    })

 })