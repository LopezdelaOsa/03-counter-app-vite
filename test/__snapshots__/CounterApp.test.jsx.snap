// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Coincidencia con snapshot 1`] = `
<div>
  <h1>
    Counter - App
  </h1>
  <h2>
    100
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="boton-reset"
  >
     Inicio 
  </button>
</div>
`;
